.github/workflows/bcny-firebase.yml

```diff
@@ -348,3 +348,137 @@ jobs:
+          export DST=${{ github.workspace }}/SourceCache/firebase-cpp-sdk
+          function cmake_build {
+              arch=$1
+              module=$2
+              extra_params=$3
+              src_dir=$SRC
+              dst_dir=$DST
+              ndk_dir="$NDK_ROOT"
+              config_command="cmake \
+          	-H$src_dir \
+          	-B$dst_dir/$module/.externalNativeBuild/cmake/release/$arch \
+          	-DANDROID_ABI=$arch \
+          	-DANDROID_PLATFORM=android-19 \
+          	-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=$dst_dir/$module/build/intermediates/cmake/release/obj/$arch \
+          	-D CMAKE_INSTALL_PREFIX=${{ github.workspace }}/BuildRoot/Library/firebase/usr \
+          	-DCMAKE_BUILD_TYPE=Release \
+          	-DANDROID_NDK=$ndk_dir \
+          	-DCMAKE_TOOLCHAIN_FILE=$ndk_dir/build/cmake/android.toolchain.cmake \
+          	-GNinja \
+          	-DFIREBASE_INCLUDE_LIBRARY_DEFAULT=OFF \
+          	-D FIREBASE_GITHUB_ACTION_BUILD=YES \
+          	$extra_params"
+              echo "Configuring $module..."
+              eval "$config_command"
+              echo "Building $module..."
+              cmake --build $dst_dir/$module/.externalNativeBuild/cmake/release/$arch
+          }
+          arch="${{ matrix.arch }}"
+          cd ${{ github.workspace }}/SourceCache/firebase-cpp-sdk
+          
+          # Prebuild the gradle app to avoid any gradle <-> cmake issues on GHA workflow.
+          ./gradlew :app:invites_resources:generateDexJarRelease
+
+          # Build the invididual components
+          cmake_build $arch "app" ""
+          cmake_build $arch "storage" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_STORAGE=ON"
+          cmake_build $arch "messaging" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_MESSAGING=ON"
+          cmake_build $arch "analytics" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_ANALYTICS=ON"
+          cmake_build $arch "database" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_DATABASE=ON"
+          cmake_build $arch "gma" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_GMA=ON"
+          cmake_build $arch "dynamic_links" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_DYNAMIC_LINKS=ON"
+          cmake_build $arch "installations" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_INSTALLATIONS=ON"
+          cmake_build $arch "app_check" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_APP_CHECK=ON"
+          cmake_build $arch "firestore" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_FIRESTORE=ON"
+          cmake_build $arch "auth" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_AUTH=ON"
+          cmake_build $arch "functions" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_FUNCTIONS=ON"
+          cmake_build $arch "remote_config" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_REMOTE_CONFIG=ON"
```

Comment:
This feels like its building more than we do on Windows, but I think that getting it built is more important than optimizing right now.